src/generator/operations.ts

```diff
@@ -575,9 +575,29 @@ function createProtocolRequest(codeModel: CodeModel, op: Operation, imports: Imp
       text += '\treturn req, nil\n';
     }
   } else if (mediaType === 'binary') {
-    const contentType = op.requests![0].protocol.http!.mediaTypes[0];
+    let contentType = `"${op.requests![0].protocol.http!.mediaTypes[0]}"`;
+    if (op.requests![0].protocol.http!.mediaTypes.length > 1) {
+      for (const param of values(op.requests![0].parameters)) {
+        // If a request defined more than one possible media type, then the param is expected to be synthesized from modelerfour
+        // and should be a SealedChoice schema type that account for the acceptable media types defined in the swagger. 
+        if (param.origin !== undefined && param.origin === 'modelerfour:synthesized/content-type' && param.schema.type === SchemaType.SealedChoice) {
```

Comment:
You should be able to omit the `undefined` check.

Reply:
done